proposals/lip-replace_static_fee_system_by_dynamic_fee_system.md

```diff
@@ -0,0 +1,358 @@
+
+## Specification
+
+Regarding the protocol, the changes involved for the dynamic fee system are the following:
+
+ - **Three constants** will define the minimum required fee as explained in the previous section<sup>[1]</sup>:
+
+    ```js
+   fees: {
+        minFeePerByte: 1000,
+        delegateFee: 1000000000,
+        dappFee: 2500000000,
+       },
+       ```
+
+- **Transactions verification**: The fixed fee logic has to be replaced in the transaction verification process to accept a dynamic fee structure:
+
+    - A transaction, `trs`, is valid if it spends a fee equal or higher than `trs.minFee`. Depending on the transaction type, `trs.minFee` is calculated as:
+
+      1. For transaction types 0, 1, 3 and 4:
+      ```
+      trs.minFee = fees.minFeePerByte * sizeof(trs)
+      ```
+      2. For transaction type 2:
+      ```
+      trs.minFee = fees.delegateFee + fees.minFeePerByte * sizeof(trs)
+      ```
+      3. For transaction type 5:
+      ```
+      trs.minFee = fees.dappFee + fees.minFeePerByte * sizeof(trs)
+      ```
+
+    - The previous point implies to have a method to calculate the size of the specific transaction in bytes: `sizeof()`. In this proposal, we define the size of a transaction, `trs`, as the size in bytes of the byte array used to generate the transaction ID (as for example it is done [here](https://github.com/LiskHQ/lisk/blob/0c633a2f0d74bcb104ae4dafc20b55ca162a0f3e/logic/block.js#L112) in the current implementation).
+
+-  **Include fee property in transaction JSON object**: With the implementation of [LIP-0012](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0012.md), the `trs.fee` property is going to be removed from the transaction JSON object since it is not needed for the current static fee system. However, this `trs.fee`, of type _BigNum_, has to be included in the transaction object, otherwise the nodes would not be able to get and verify the transaction fee.
+
+- **Include fee property in transaction signature and transaction ID**: To prevent transaction fee forging and manipulation by malicious nodes/delegates, the transaction fee has to be part of the byte array used as input for the transaction signature and transaction ID generation (e.g. implemented by [`getBytes`](https://github.com/LiskHQ/lisk/blob/5d5af525eec7f2ee687ca921a6a2d1457c471da9/logic/transaction.js#L154) in Lisk core). The `trs.fee` property, with a size of 8 bytes and encoded in big-endian, has to follow the `trs.senderPublicKey` property during serialisation. Additionally, the [`getTransactionBytes`](https://github.com/LiskHQ/lisk-elements/blob/d5ab830ad65965764d364eb344ce4e3a13bf14c9/packages/lisk-transactions/src/utils/get_transaction_bytes.js#L147) function must be implemented in the same way in Lisk Elements.
+
```

Comment:
I would say `as input for the transaction signature generation` or `as input for the transaction signing`.

Reply:
changed to the first option